lib/vm.js

```diff
@@ -286,6 +287,49 @@ function runInThisContext(code, options) {
   return createScript(code, options).runInThisContext(options);
 }
 
+function compileFunction(code, params, options) {
+  const defaults = {
+    filename: '',
+    columnOffset: 0,
+    lineOffset: 0,
+    cachedData: undefined,
+    produceCachedData: false,
+    parsingContext: undefined
+  };
+  const {
+    filename,
+    columnOffset,
+    lineOffset,
+    cachedData,
+    produceCachedData,
+    parsingContext,
+  } = Object.assign(defaults, options);
+
+  if (typeof code !== 'string') {
+    throw new ERR_INVALID_ARG_TYPE('code', 'string', code);
+  }
+  if (Array.isArray(params)) {
```

Comment:
You can save a level of indentation here if you check for `!Array.isArray(params)` with a `throw`, and drop the `else`.

Reply:
Done.